packages/databricks-vscode/src/configuration/models/ConfigModel.ts

```diff
@@ -208,35 +208,42 @@ export class ConfigModel implements Disposable {
         ) {
             throw new Error(`Target '${target}' doesn't exist in the bundle`);
         }
-        await this.configsMutex.synchronise(async () => {
-            this._target = target;
-            await this.stateStorage.set("databricks.bundle.target", target);
-            // We want to wait for all the configs to be loaded before we emit any change events from the
-            // configStateCache.
-            await this.readStateMutex.synchronise(async () => {
+
+        try {
+            await this.configsMutex.synchronise(async () => {
+                this._target = target;
+                await this.stateStorage.set("databricks.bundle.target", target);
+                // We want to wait for all the configs to be loaded before we emit any change events from the
+                // configStateCache.
+                this.bundlePreValidateModel.setTarget(target);
+                this.bundleValidateModel.setTarget(target);
+                this.overrideableConfigModel.setTarget(target);
+                this.bundleRemoteStateModel.setTarget(target);
+                this.onDidChangeTargetEmitter.fire();
```

Comment:
I wonder if we now have a race condition because you emit the target change before refreshing the models. During the refresh the connection manager tries to login with saved auth and might use stale values from the configModel (since its refresh it tied to the refreshes of other models, which might not happen in time)

Reply:
Should we emit the target change at the very end of the method?